pkg/pipeline/processing_test.go

```diff
@@ -0,0 +1,289 @@
+
+func (p PanicProcessing) Process(ctx context.Context, data Data) error {
+	panic(panicReason)
+}
+
+// Test
+
+func TestProcessingHelpers(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Processing helpers test suite")
+}
+
+// Test Parallel
+
+var _ = Describe("Testing ParallelProcessing with 2 Processing", func() {
+	var ctrl *gomock.Controller
+
+	var parallel pipeline.Processing[Data]
+	var proc1, proc2 *mock.MockProcessing[Data]
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+
+		proc1 = mock.NewMockProcessing[Data](ctrl)
+		proc2 = mock.NewMockProcessing[Data](ctrl)
+
+		parallel = pipeline.NewParallelProcessing(proc1, proc2)
+	})
+
+	When("both processing return nil", func() {
+		BeforeEach(func() {
+			proc1.EXPECT().Process(gomock.Any(), data).Return(nil).Times(1)
+			proc2.EXPECT().Process(gomock.Any(), data).Return(nil).Times(1)
+		})
+
+		It("should succeed", func(ctx SpecContext) {
+			By("calling Process")
+			err := parallel.Process(ctx, data)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	When("only the first processing returns an error", func() {
+		Context("and the error is a retryable ErrProcessingError", func() {
+			BeforeEach(func() {
+				proc1.EXPECT().Process(gomock.Any(), data).Return(errRetryableErrProcessingError).Times(1)
+				proc2.EXPECT().Process(gomock.Any(), data).Return(nil).Times(1)
+			})
+
+			It("should return a retryable ErrProcessingError", func(ctx SpecContext) {
+				err := parallel.Process(ctx, data)
+
+				Expect(err).To(HaveOccurred(), "nil error")
+
+				Expect(errors.Is(err, pipeline.ErrRetryableError)).To(BeTrue(), "error is retryable")
+
+				processingError := pipeline.ErrProcessingError{}
+				Expect(errors.As(err, &processingError)).To(BeTrue(), "error is a ErrProcessingError")
+				Expect(processingError.Category).To(Equal(oneCategory), "ErrProcessingError category is preserved")
+			})
+		})
+
+		Context("and the error is generic", func() {
+			BeforeEach(func() {
+				proc1.EXPECT().Process(gomock.Any(), data).Return(errOneError).Times(1)
+				proc2.EXPECT().Process(gomock.Any(), data).Return(nil).Times(1)
+			})
+
+			It("should fail", func(ctx SpecContext) {
+				err := parallel.Process(ctx, data)
+				Expect(err).To(HaveOccurred(), "non nil error")
+				Expect(errors.Is(err, errOneError)).To(BeTrue(), "error is the original error")
+			})
+		})
+	})
+
+	When("only the second processing returns an error", func() {
+		Context("and the error is a retryable ErrProcessingError", func() {
+			BeforeEach(func() {
+				proc1.EXPECT().Process(gomock.Any(), data).Return(nil).Times(1)
+				proc2.EXPECT().Process(gomock.Any(), data).Return(errRetryableErrProcessingError).Times(1)
+			})
+
+			It("should return a retryable ErrProcessingError", func(ctx SpecContext) {
+				err := parallel.Process(ctx, data)
+
+				Expect(err).To(HaveOccurred(), "nil error")
+
+				Expect(errors.Is(err, pipeline.ErrRetryableError)).To(BeTrue(), "error is retryable")
```

Comment:
I think we can use `MatchError` instead of all the `errors.Is`:
```suggestion
				Expect(err).Should(MatchError(pipeline.ErrRetryableError))
```